v-next/hardhat-ledger/test/internal/handler.ts

```diff
@@ -0,0 +1,813 @@
+        });
+
+        ledgerHandler = new LedgerHandler(
+          ethereumMockedProvider,
+          {
+            accounts: LEDGER_ADDRESSES,
+          },
+          mockedDisplayInfo.fn,
+          {
+            ethConstructor: eth,
+            transportNodeHid: getTransportNodeHidMock(),
+            cachePath: tmpCachePath,
+          },
+        );
+
+        const request = createJsonRpcRequest("eth_signTypedData_v4", [
+          account.address,
+          typedMessage,
+        ]);
+
+        const res = await ledgerHandler.handle(request, []);
+
+        assert.ok(res !== null, "res should not be null");
+        assert.ok("result" in res, "res should have the property 'result'");
+
+        assert.deepEqual(res.result, signature);
+      });
+
+      it("should successfully handle the method eth_sendTransaction", async () => {
+        const signedRawTx =
+          "0xf861646482520894da6a52afdae5ff66aa786da68754a227331f56e3648082f4f5a04ab14d7e96a8bc7390cfffa0260d4b82848428ce7f5b8dd367d13bf31944b6c0a03cc226daa6a2f4e22334c59c2e04ac72672af72907ec9c4a601189858ba60069";
+
+        const txRsv = {
+          v: "f4f5",
+          r: "4ab14d7e96a8bc7390cfffa0260d4b82848428ce7f5b8dd367d13bf31944b6c0",
+          s: "3cc226daa6a2f4e22334c59c2e04ac72672af72907ec9c4a601189858ba60069",
+        };
+
+        [eth] = getEthMocked({
+          getAddress: {
+            result: (searchedPath: string) => {
+              return searchedPath === derPath
+                ? account
+                : { address: "0x0", publicKey: "0x0" };
+            },
+          },
+          signTransaction: {
+            result: txRsv,
+            expectedParams: {
+              path: derPath,
+              rawTxHex:
+                "e1646482520894da6a52afdae5ff66aa786da68754a227331f56e36480827a698080",
+              resolution: {
+                nfts: [],
+                erc20Tokens: [],
+                externalPlugin: [],
+                plugin: [],
+                domains: [],
+              },
+            },
+          },
+        });
+
+        ethereumMockedProvider.setReturnValue("eth_chainId", "0x7a69");
+        ethereumMockedProvider.setReturnValue(
+          "eth_getTransactionCount",
+          "0x64",
+        );
+
+        ledgerHandler = new LedgerHandler(
+          ethereumMockedProvider,
+          {
+            accounts: LEDGER_ADDRESSES,
+          },
+          mockedDisplayInfo.fn,
+          {
+            ethConstructor: eth,
+            transportNodeHid: getTransportNodeHidMock(),
+            cachePath: tmpCachePath,
+          },
+        );
+
+        const request = createJsonRpcRequest("eth_sendTransaction", [
+          {
+            from: account.address,
+            to: LEDGER_ADDRESSES[1],
+            value: numberToHexString(100),
+            gas: numberToHexString(1000000),
+            gasPrice: numberToHexString(100),
+            gasLimit: numberToHexString(1000000),
+          },
+        ]);
```

Comment:
When I alter the `gas` field the test still passes. I am not sure that all of the gas related fields are going through.

We should double check that the signed transaction isn't just ending up with the defaults from `micro-eth-signer`.

Can we add another test here, so one for:
* pre-eip 1559 transactions with the appropriate gas fields
* post-eip 1559 transactions with teh appropriate gas fields